src/site/shared/containers/HistoryBox/index.tsx

```diff
@@ -0,0 +1,62 @@
+import {CircuitInfo} from "core/utils/CircuitInfo";
+
+import {useSharedDispatch, useSharedSelector} from "shared/utils/hooks/useShared";
+import {useHistory} from "shared/utils/hooks/useHistory";
+
+import "./index.scss";
+import {Action} from "core/actions/Action";
+import {GroupAction} from "core/actions/GroupAction";
+
+
+type HistoryEntryProps = {
+    a: Action;
+}
+const HistoryEntry = ({a}: HistoryEntryProps) => {
+    if (a instanceof GroupAction) {
+        return(<GroupActionEntry g = {a}></GroupActionEntry>)
+    }
+    return(<div className="historybox__entry">{a.getName()}</div>);
+}
+
+
+type GroupActionEntryProps = {
+    g: GroupAction;
+}
+const GroupActionEntry = ({g}: GroupActionEntryProps) => {
+    if (g.isEmpty()) {
+        return null;
+    }
+    else if (g.getActions().length == 1) {
```

Comment:
nit: use just `if` again since the previous `if` returns if true, and then also remove braces
nit: use `===`